src/mailroom.py

```diff
@@ -0,0 +1,169 @@
+# -*- coding: utf-8 -*-
+""" This is the mailroom madness script. It will run from the cmd line and
+ allow you to generate an email or report based on user input."""
+
+import os
+import sys
+import math
+
+
+DONOR_DICT = {'Linus Torvalds': [100, 150, 250],
+              'Jane Goodall': [200, 250, 50],
+              'Bob Ross': [1000, 550, 1]}
+
+
+def main_menu_input():
+    '''Display user prompt for main menu'''
+    user_input = input('''
+    Main Menu:
+    E to generate a thank you email to send to a donor
+    R for a report on the donation history of all donors
+    Type exit at any prompt to exit the program.
+
+    >>>''').upper()
+    menu_input_validation(user_input)
+
+
+def email_donor_input():
+    user_input = input('''
+    Please enter in a donor name.
+    Type list for a list of all donors.
+    Type menu to return to the main menu prompt.
+
+    >>>''').upper()
+    email_input_validation(user_input)
+
+
+def email_donation_input():
+    user_input = int(input('''
+    Please enter the donation amount in numbers only.
+    >>>'''))
+    if type(user_input) == int or type(user_input) == float:
+        return float(user_input)
+    else:
+        email_donation_input()
+
+
+def menu_input_validation(user_input):
+    '''Validate user input from the main menu'''
+    if user_input != 'EXIT' and user_input != 'E' and user_input != 'R':
+        print('\nInvalid input. A valid input is X, E or R.\n')
+        main_menu_input()
+    else:
+        main_menu_route(user_input)
```

Comment:
Imagine that this returned either the value of `user_input` (if valid) or None (if the input was not valid).

Then imagine a structure like this:

```python
user_input = None
while user_input is None:
    user_input = menu_input_validation(main_menu_input())
main_menu_route(user_input)
```
You get the same basic effect as what you've created, but all three of these functions are now independent of each-other, and can be tested.